### PR TITLE
Android app changes p2p

### PR DESCRIPTION
**Pull request by mgcouchbase**
Refactored Android, Java Testserver App code using the context style.

Java- WS app build:
     http://mobile.jenkins.couchbase.com/view/CBL%202.x/job/couchbase-lite-java-webservice-testapp/60/console
Java- DS App build:
     http://mobile.jenkins.couchbase.com/view/CBL%202.x/job/couchbase-lite-java-desktop-testapp/62/
Android:
    http://mobile.jenkins.couchbase.com/view/CBL%202.x/job/couchbase-lite-android-testapp/948/console
http://uberjenkins.sc.couchbase.com:8080/view/testkit-testing/job/android-p2p/3/testReport/

new updated and ran WS build
http://mobile.jenkins.couchbase.com/view/CBL%202.x/job/couchbase-lite-java-webservice-testapp/61/console

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-Android/app/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apply plugin: 'com.android.application'
 
 ext {
-    DEF_VERSION = "2.8.0-204"
+    DEF_VERSION = "2.8.0-267"
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Android/app/build.gradle:4]: revert this change, we do need to add latest version here

**Review by sridevi-15 — COMMENTED** [review]
manasa, Changes are okay except for reverting the version change.
looks like lot of test app changes which might effect the current Android and java apps. Can you provide jenkins link with successful app created on Android and java and have one p2p test jenkins run on Android and java?

This is for safe side to avoid breaking master

File: CBLClient/Apps/CBLTestServer-Android/app/build.gradle (hunk 1 above)
Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-Android/app/build.gradle]: done

**Comment by euniceatcouchbase** [comment]
this PR looks very messy. all common java code changes have be included in the java p2p PR, you don't have to repeat here. and I would prefer you leave along those gradle/gradlew files. and the c# file shouldn't be in this PR either.

Before you send a PR, you may diff your origin branch with origin/master, preview the change lists and changes on you local environment. You will want to minimize the line of changes.

**Comment by mgcouchbase** [comment]
Cleaned the PR

File: CBLClient/Apps/CBLTestServer-Java-WS/src/main/java/com/couchbase/javaws/TestServerContext.java [hunk 2]
```diff
@@ -1,12 +1,36 @@
-package com.couchbase.javaws;
-
+package com.couchbase.mobiletestkit.javatestserver;
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-Java-WS/src/main/java/com/couchbase/javaws/TestServerContext.java:1]: java ties file structure with package name. if you make this change, you have to move the file from javws directory to the new directory you specified in package name, and other package reference in entire project will have to be updated too. I would suggest you not touch the package name.

**Review by euniceatcouchbase — CHANGES_REQUESTED** [review]

**Review by euniceatcouchbase — APPROVED** [review]

**Review by euniceatcouchbase — APPROVED** [review]

**Review by sridevi-15 — APPROVED** [review]
